### PR TITLE
add channel context to the OrderItemOptionFactory

### PR DESCRIPTION
**Pull request by t-n-y**

### CODE REVIEW

**Comment by mamazu** [comment]
How can that be that the order doesn't have a channel? And using the channel context will break creating an order from the admin panel.

File: src/Factory/OrderItemOptionFactory.php [hunk 1]
```diff
@@ -63,8 +65,9 @@ public function createForOptionAndValue(
             /** @var ProductInterface $product */
             $product = $orderItem->getProduct();
 
+            $contextChannel = $this->channelContext->getChannel();
             /** @var ChannelInterface $channel */
-            $channel = $order->getChannel();
+            $channel = $order ? $order->getChannel() : $contextChannel;
```
Comment (seizan8) [inline src/Factory/OrderItemOptionFactory.php:70]: @t-n-y how do you get orderItems without order? I don't see a case where order wouldn't be set on an orderItem

Comment (t-n-y) [reply src/Factory/OrderItemOptionFactory.php:70]: I don't know neither. I don't understand how it is possible.
i opened an issue with my probleme [https://github.com/Brille24/SyliusCustomOptionsPlugin/issues/146](https://github.com/Brille24/SyliusCustomOptionsPlugin/issues/146)

**Comment by t-n-y** [comment]
@mamazu @seizan8 I don't know neither. I don't understand how it is possible.
i opened an issue with my problem https://github.com/Brille24/SyliusCustomOptionsPlugin/issues/146

**Comment by t-n-y** [comment]
@seizan8 i understand. 
tell me if you can reproduce this error or if you need more explanations

**Comment by seizan8** [comment]
@t-n-y sorry man, I currently have no time to look into this

**Comment by t-n-y** [comment]
@seizan8 no pb, same for me. will see if i can find some time in the next days

**Comment by JoppeDC** [comment]
@t-n-y Just for info, I've been able to reproduce this issue by using a `Select` option on a product

**Pull request closed without merging**